### PR TITLE
Qt6 compatibility upgrade

### PR DESCRIPTION
**Pull request by gzotti**
This branch should be used as we adapt to Qt6.

Milestones:

- [x] CMakeLists.txt  should work with Qt5.12 and later. Recommended: Qt5.15.2 and Qt6.2.4 (or later)
- [ ] external modules should be removed where possible, upgraded to current where required, or fixed. 
- [x] esp. qcustomplot does not compile. Get rid of it (use QCharts; #2390, #2447).
- [ ] Replace QtScript by QJSEngine functions (#2461)
- [ ] Upgrade libindi (or better, remove source duplication) so that it compiles with latest gcc (#2157)
- [x] **Compile main app (without plugins) with Qt6**
- [x] **Restore broken text everywhere**
- [x] Compile all plugins with Qt6
- [ ] Upgrade StelAudioMgr module (compiles, but may need fixes with API change) --> Completion after reactivating Scripting in Qt6!
- [ ] Upgrade StelVideoMgr module (compiles, but may need fixes with API change) --> Completion after reactivating Scripting in Qt6!
- [x] **Restore Scenery3D rendering** (OBJ loader broken?)

To build with Qt6, disable SCRIPTING. This is dealt with in another branch  (#2461).

I think the best strategy for now is working on master with Qt5.15+, trying to get rid of deprecated stuff hinted at by clazy, use feature branches for larger tasks, and occasionally rebase this branch from master to test.

### Update 2022-06-08
The program builds! However, text is garbled. 
### Update 2022-06-29
Finally! Text is visible again. 

It should be possible to keep compilation compatible between Qt5 and Qt6, without any extra branch. Qt6 requires 64bit CPUs and fully working OpenGL graphics, so Qt5 will likely have to stay around for a few more years. I have re-extended compatibility back to Qt5.12 (Ubuntu 20LTS, and even 5.9, for Ubuntu 18LTS). 

### Description

Fixes # (issue)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

### How Has This Been Tested?

**Test Configuration**
***Operating system: Windows 10***
* Graphics Card: Nvidia Geforce 960M
* Qt5.9.9/MSVC2017 configuration with MSVC2019. 
* Qt5.12.8/MSVC2017 configuration with MSVC2019. 
* Qt5.15.2/MSVC2019 configuration with MSVC2017 and MSVC2019. 
* Qt6.2.4/MSVC2019 configuration with MSVC2019. Qt6 really requires MSVC2019!

***Operating system: Ubuntu 22.04***
* Graphics Card: (Intel, whatever built into i3-5010U)
* Qt5.15.3 
* Qt6.2.4

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [x] I have updated the respective chapter in the Stellarium User Guide
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by gzotti** [comment]
Dear all, 

I did not play with the version numbers, but this may be Stellarium 1.0 alpha.

```
2022-06-08T17:10:26
Operating System: Windows 10 Version 2009
Compiled using MSVC++ 16.10, 16.11 (Visual Studio 2019)
Qt runtime version: 6.2.4
Qt compilation version: 6.2.4
Addressing mode: 64-bit
Total physical memory: 32631 MB
Available physical memory: 10046 MB
Physical memory in use: 69%
Processor speed: 2592 MHz
Processor name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
Processor speed: 2592 MHz
Processor name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
Processor speed: 2592 MHz
Processor name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
Processor speed: 2592 MHz
Processor name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
Processor speed: 2592 MHz
Processor name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
Processor speed: 2592 MHz
Processor name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
Processor speed: 2592 MHz
Processor name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
Processor speed: 2592 MHz
Processor name: Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
D:\StelDev\GIT\build-stellarium-Desktop_Qt_6_2_4_MSVC2019_64bit-Release\src\stellarium.exe -platform windows:altgr 
 ---------------------------------------------------------------------------
[ This is Stellarium 0.22.1-532a784 [towards-Qt6] - https://stellarium.org/ ]
[ Copyright �� 2000-2022 Stellarium Developers                              ]
 ---------------------------------------------------------------------------

```

I will push this soon. However, all texts are squashed to the left of each window, so there is still work ahead. Also 2 plugins do not yet compile.
grafik

**Comment by gzotti** [comment]
Sorry, this ugly monstrosity of a merge was caused by editing one conflict online.

**Comment by gzotti** [comment]
I have seriously no idea what's going on with the failing text rendering. @alex-w @xalioth @Atque Any ideas?

**Comment by alex-w** [comment]
> I have seriously no idea what's going on with the failing text rendering. @alex-w @xalioth @Atque Any ideas?

On macOS 12.4 with Qt 6.3.0 (official package from Qt.io) the main scene is not render (just a black screen) :(

**Comment by gzotti** [comment]
Sad. From what I am reading, Mac is leaving OpenGL, replacing by Metal, and Windows still pushes DirectX. Qt6 lost ANGLE, so only full OpenGL PCs will be able to run the Qt6-based series (or MESA should still be available, but of course less fun), unless some GPU hero does the new Qt6 graphics subsystem glue code. .

I cannot find any reports online about bad font rendering or similar terms. This is maddening.

**Comment by gzotti** [comment]
I followed a single relevant hint from Stackexchange, where sb describes the same problem in a QML application. Now we can at least see text when using a monospaced font like Courier. 
grafik

However, this is not a complete solution. With proportional fonts it looks terrible.

grafik


What is wrong with the app initialisation so that font metrics believe character width is generally Zero? There are differences between Qt5 and Qt6 results (for now described in main.cpp:L384ff).

**Comment by gzotti** [comment]
My WSL is still Ubuntu 20.04LTS, i.e. no Qt6. Is it safe to upgrade to 22.04.0 already? (the 22.04.1 which will upgrade 20.04LTS comes in August only.)

**Comment by alex-w** [comment]
> My WSL is still Ubuntu 20.04LTS, i.e. no Qt6. Is it safe to upgrade to 22.04.0 already? (the 22.04.1 which will upgrade 20.04LTS comes in August only.)

Oops... I have U20.04 on the my desktop, so, I don't have an answer :-/ We have an special case for reading and set fonts for Windows + we load and use an own fonts and it may have some specifics in Qt6.

**Comment by gzotti** [comment]
Upgrading one system to U22.04...

**Comment by alex-w** [comment]
This port is not useable on macOS :(

See https://bugreports.qt.io/browse/QTBUG-100686 for example

**Comment by gzotti** [comment]
> According to [AppVeyor](https://ci.appveyor.com/project/alex-w/stellarium/builds/44046555) Stellarium@Qt6 is very slow building and CI is killing the process of building :( Of course we can disable unit testing for Qt6, but this is not very good. Currently we cannot to obtain the binary packages for testing Qt6-based Stellarium.
> 
> @gzotti any ideas for speed-up the building?

As we noted before, translation takes ages. Usually I switch off NLS for developing. The result is just English. Should be OK for an early beta.
I hope you have configured SCRIPTING off for the current state? (Should be done by cmake...)

**Comment by alex-w** [comment]
> > This port is not useable on macOS :(
> > See https://bugreports.qt.io/browse/QTBUG-100686 for example
> 
> At least "not yet". I guess the "black window" may be caused by the obsolescence of OpenGL on MacOS. Then MacOS users can keep using Qt5-based versions until somebody finds out what has to be done with their shiny "Metal".

The "black window" I see with official Qt5.15 and Qt6. When I use Qt5 from homebrew, then I see "good window". And I cannot build the binary package homebrew-Qt6-based Stellarium on macOS due bug in macosdeploy component.

**Comment by gzotti** [comment]
> The "black window" I see with official Qt5.15 and Qt6. When I use Qt5 from homebrew, then I see "good window". And I cannot build the binary package homebrew-Qt6-based Stellarium on macOS due bug in macosdeploy component.

Weird. I cannot help with Mac issues, though. :-(

**Comment by gzotti** [comment]
> > As we noted before, translation takes ages. Usually I switch off NLS for developing. The result is just English. Should be OK for an early beta. I hope you have configured SCRIPTING off for the current state? (Should be done by cmake...)
> 
> NLS=Off, TESTING=On, SCRIPTING=Off for Qt6.... the process of building taken over 45 minutes

No, that's too slow. Should not be worse than 5.15.

**Comment by alex-w** [comment]
> > NLS=Off, TESTING=On, SCRIPTING=Off for Qt6.... the process of building taken over 45 minutes
> 
> No, that's too slow. Should not be worse than 5.15.

See partially success build: https://ci.appveyor.com/project/alex-w/stellarium/builds/44048756 (unit tests fails)

**Comment by alex-w** [comment]
The "black window" log:
> qt.qpa.openglcontext: Could not create NSOpenGLContext with shared context, falling back to unshared context.
> UNSUPPORTED (log once): POSSIBLE ISSUE: unit 0 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable

I fear the wrapper on macOS cannot convert OpenGL code, because our OpenGL code is old (lesser 3.0) and Qt expect at least 3.0+

**Comment by gzotti** [comment]
Hehe, years ago in early Qt5 times one from the first team ranted about my "stupid OpenGL test and diagnosis code" which was unnecessary on MacOS. I concluded "OpenGL never fails on Mac" and excluded Macs from the test/diagnostic output, and never again cared for it. 

We initialize asking for an OpenGL 2.1 context. On PCs, the driver is free to serve a better one, but allegedly on Mac you get what you ask for. My systems deliver OpenGL 4.3 or 4.6 contexts. We probably could exclude any ANGLE related code from Qt6. But I am not sure how SBCs behave when asking for some GLES version. I could probably set up a Raspi4 with U22.04 to check what OpenGL flavour is detected.

We have a command-line option -compat33. Does that do anything? Or can it be fixed to do something useful?

**Comment by gzotti** [comment]
> Qt 5.12:
> 
> ```
> initializeGL
> OpenGL supported version:  "2.1 Metal - 76.3"
> Current Format:  QSurfaceFormat(version 2.1, options QFlags<QSurfaceFormat::FormatOption>(DeprecatedFunctions), depthBufferSize 32, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize 8, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::DoubleBuffer, swapInterval 0, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::NoProfile)
> StelMainView::init
> Driver version string: "2.1 Metal - 76.3"
> GL vendor is "Apple"
> GL renderer is "Apple M1"
> GL Shading Language version is "1.20"
> ```

Interesting. We know for years that we need 1.30 on Linux and Windows.

**Comment by gzotti** [comment]
I see a new warning:

> Warning: Setting a new default format with a different version or profile after the global shared context is created may cause issues with context sharing.

It seems we must set default format before even initializing the qApp object. This means, configuration must be processed in main.cpp, even before CLIprocessor, and config via CLI options are probably impossible. For whatever reason the --compat33 switch does currently nothing, although creating a 3.3 compat profile seems to work.

**Comment by gzotti** [comment]
@alex-w this now should request a 3.3 compatibility profile on MacOS. Does that help preventing the black window? (Probably the later re-setting of format should be suppressed?) If Macs are guaranteed to run at least OpenGL 3.3, we could just force that.

**Comment by alex-w** [comment]
> @alex-w this now should request a 3.3 compatibility profile on MacOS. Does that help preventing the black window? (Probably the later re-setting of format should be suppressed?) If Macs are guaranteed to run at least OpenGL 3.3, we could just force that.

I see "good window" now:
```
Default surface format:  QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize 8, stencilBufferSize 8, samples -1, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::NoProfile)
Desired surface format:  QSurfaceFormat(version 2.1, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize 8, stencilBufferSize 8, samples -1, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 0, colorSpace QColorSpace(), profile  QSurfaceFormat::NoProfile)
setDefaultFormat
Warning: Setting a new default format with a different version or profile after the global shared context is created may cause issues with context sharing.
Creating GL widget
StelGLWidget constructor
setViewport
done
StelGraphicsScene constructor
initializeGL
OpenGL supported version:  "2.1 Metal - 76.3"
Current Format:  QSurfaceFormat(version 2.1, options QFlags<QSurfaceFormat::FormatOption>(DeprecatedFunctions), depthBufferSize 32, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize 8, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::DoubleBuffer, swapInterval 0, colorSpace QColorSpace(), profile  QSurfaceFormat::NoProfile)
StelMainView::init
Driver version string: "2.1 Metal - 76.3"
GL vendor is "Apple"
GL renderer is "Apple M1"
GL Shading Language version is "1.20"
Detected a high resolution device! Device pixel ratio: 2
```

In the CLI I see also:
```
An OpenGL surfcace format was requested that is either not version 3.2 or higher or a not Core Profile.
Chromium on macOS will fall back to software rendering in this case.
Hardware acceleration and features such as WebGL will not be available.
```

**Comment by gzotti** [comment]
No,that's still not good. Do you know the minimum OpenGL version Apple's MacOS systems are guaranteed to deliver? 3.3? or even 4.0? Set this in main.cpp line 198ff. Omit the line 201 (setting compatibility profile). And probably modify StelMainView::getDesiredGLFormat() to not change version numbers in MacOS.

**Comment by alex-w** [comment]
> No,that's still not good. Do you know the minimum OpenGL version Apple's MacOS systems are guaranteed to deliver? 3.3? or even 4.0? Set this in main.cpp line 198ff. Omit the line 201 (setting compatibility profile). And probably modify StelMainView::getDesiredGLFormat() to not change version numbers in MacOS.

See the part of CLI:
```
An OpenGL surfcace format was requested that is either not version 3.2 or higher or a not Core Profile.
Chromium on macOS will fall back to software rendering in this case.
Hardware acceleration and features such as WebGL will not be available.
 ---------------------------------------------------------------------------
[ This is Stellarium 1.22.1-9f75b99 [towards-Qt6] - https://stellarium.org/ ]
[ Copyright (C) 2000-2022 Stellarium Developers                             ]
 ---------------------------------------------------------------------------
Writing log file to: "/Users/aw/Library/Application Support/Stellarium/log.txt"
File search paths:
  0 .  "/Users/aw/Library/Application Support/Stellarium"
  1 .  "/Users/aw/stellarium"
Config file is:  "/Users/aw/Library/Application Support/Stellarium/config.ini"
Default surface format:  QSurfaceFormat(version 2.0, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize 8, stencilBufferSize 8, samples -1, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 1, colorSpace QColorSpace(), profile  QSurfaceFormat::NoProfile)
Desired surface format:  QSurfaceFormat(version 3.3, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize 8, stencilBufferSize 8, samples -1, swapBehavior QSurfaceFormat::DefaultSwapBehavior, swapInterval 0, colorSpace QColorSpace(), profile  QSurfaceFormat::CompatibilityProfile)
Creating GL widget
StelGLWidget constructor
The viewport is settled...
StelGraphicsScene constructor
initializeGL
OpenGL supported version:  "2.1 Metal - 76.3"
Current Format:  QSurfaceFormat(version 2.1, options QFlags<QSurfaceFormat::FormatOption>(DeprecatedFunctions), depthBufferSize 32, redBufferSize 8, greenBufferSize 8, blueBufferSize 8, alphaBufferSize 8, stencilBufferSize 8, samples 0, swapBehavior QSurfaceFormat::DoubleBuffer, swapInterval 0, colorSpace QColorSpace(), profile  QSurfaceFormat::NoProfile)
StelMainView::init
Driver version string: "2.1 Metal - 76.3"
GL vendor is "Apple"
GL renderer is "Apple M1"
GL Shading Language version is "1.20"
Detected a high resolution device! Device pixel ratio: 2
```

**Comment by gzotti** [comment]
I think we can merge this on our way to using Qt6. I compiled (and launched) with Qt5.9, 5.12, 5.15 and 6.2.4/6.3.1 on Windows.

I cannot test VideoMgr and AudioMgr with Qt6 as they are only used with the scripting engine, so this must be fixed after the QJSEngine PR (#2461). I changed the relevant calls so that they at least compile. 

There is a change around OpenGL initialisation (in main and StelMainView) which we can also do later. It may break ANGLE CLI options. I also don't know if Apples can be initialized to use an OpenGL 3.3 Compatibility profile.

**Comment by alex-w** [comment]
Oops… unit tests are failing with Qt6 :(

**Comment by alex-w** [comment]
@gzotti, I think we should introduce the public and private (or, better term - "inner") versions for this PR and later releases by technical reasons - GH may assign to the release only one tag, so, we cannot publish 0.22.3 & 1.22.3 under same scheme of collection tag/release as before. Plus only one release can be latest - so, we cannot publish 2 releases - 0.22.3 & 1.22.3 as latest in same time. Of course we can mark 1.22.3 as "pre-release" and maybe it can help.

**Review by alex-w — APPROVED** [review]
The pull request is almost OK for me - except issue for unit test and low performance of building on Windows.

**Comment by gzotti** [comment]
The unit test is indeed a problem. I guess you know the Octahedron* classes about as "well" as I do... What are the implications or further consequences of the difference? What breaks then, really?

**Comment by gzotti** [comment]
Abut slow building, this is I think still the lconvert issue. It has become better with Qt5.15.2, but versions before 0.15 are faster.

**Comment by gzotti** [comment]
> @gzotti, I fear we should introduce the public and private (or, better term - "inner") versions for this PR and later releases by technical reasons - GH may assign to the release only one tag, so, we cannot publish 0.22.3 & 1.22.3 under same scheme of collection tag/release as before. Plus only one release can be latest - so, we cannot publish 2 releases - 0.22.3 & 1.22.3 as latest in same time. Of course we can mark 1.22.3 as "pre-release" and maybe it can help.
> 
> P.S. This is question for discussion at the moment, but we should resolve it before release.
> 
> P.P.S. Obvious proposal for numbering of a public version (a backward compatible scheme) - two last digits of current year and number of release in the year (YY.N). Should we omitted number 0 for the release number?

Probably we must omit the prefixed 0/1 from the official release number. So, it will be 22.3. (Many people refer to year.release already.) But still, any 32bit versions must be Qt5-based, and probably the RaspberryOS or other ARM packages as well (but this may be work of Debian/Ubuntu packagers?). Can this be coded in any version number? Or does it indeed not matter? The logfile says "compiled with.../running on...", and both series are almost identical (apart from tiny details in scripting). A scripting function may have to be added to discern old/new.

**Comment by gzotti** [comment]
> 36 minutes with -DENABLE_NLS=Off and without same parameter

Strange. What hardware is that? I have a complete build (with NLS) in about 20 minutes on my 2016 notebook (i7-6xxx). Is it a VM with e.g. limited RAM?

**Comment by alex-w** [comment]
> Probably we must omit the prefixed 0/1 from the official release number. So, it will be 22.3. (Many people refer to year.release already.) But still, any 32bit versions must be Qt5-based, and probably the RaspberryOS or other ARM packages as well (but this may be work of Debian/Ubuntu packagers?). Can this be coded in any version number? Or does it indeed not matter? 

It should be not matter for packages itself, this is matter to distribution of packages

> The logfile says "compiled with.../running on...", and both series are almost identical (apart from tiny details in scripting). A scripting function may have to be added to discern old/new.

The log file (and About) will say both - public and inner version of Stellarium. I guess the splash screen should contain public version only (without name of branch?) I'll prepare patch for "new" version after merge this PR (the later of August or early September, probably). This problem is touches me most as a maintainer.

**Comment by alex-w** [comment]
> > 36 minutes with -DENABLE_NLS=Off and without same parameter
> 
> Strange. What hardware is that? I have a complete build (with NLS) in about 20 minutes on my 2016 notebook (i7-6xxx). Is it a VM with e.g. limited RAM?

Good question! The answer is... I don't know (at least I cannot find the hardware info)

**Comment by alex-w** [comment]
> The unit test is indeed a problem. I guess you know the Octahedron* classes about as "well" as I do... What are the implications or further consequences of the difference? What breaks then, really?

This is strange, but the tests passed with Qt5 - maybe some methods just changed in Qt6?

**Comment by gzotti** [comment]
> > The unit test is indeed a problem. I guess you know the Octahedron* classes about as "well" as I do... What are the implications or further consequences of the difference? What breaks then, really?
> 
> This is strange, but the tests passed with Qt5 - maybe some methods just changed in Qt6?
> 
> P.S. What about your local environment - all tests passed or not?

Yes, Qt5: all fine, Qt6: Fails 2 tests. But the SphericalGeometry & friends classes did not change! It must be something in QVector which is now the same as QList. Maybe some Qt method now behaves differently. (https://www.qt.io/blog/qlist-changes-in-qt-6)

**Comment by gzotti** [comment]
> The log file (and About) will say both - public and inner version of Stellarium. I guess the splash screen should contain public version only (without name of branch?) 

The "build" should IMHO include the inner version and branch/git SHA1 as it is now. Releases should bear [01].YY.N. But we may want to add a larger number to the splash which is the "public" number and just says YY.N. This can be overridden by configuration in important cases like "1.0" :-) Note that 0.22.2 < 1.0 < 22.4, so any distribution release scheme should be happy. 

> I'll prepare patch for "new" version after merge this PR (the later of August or early September, probably). This problem is touches me most as a maintainer.

I hope we can identify&fix the test issue this week and merge ASAP! First Betas should appear in early August (maybe without scripting if that takes longer to consolidate).

**Comment by alex-w** [comment]
see unit tests on linux & macOS (qt6)

**Comment by alex-w** [comment]
@gzotti see build time at https://ci.appveyor.com/project/alex-w/stellarium/builds/44216795 :(

**Comment by gzotti** [comment]
> @gzotti see build time at https://ci.appveyor.com/project/alex-w/stellarium/builds/44216795 :(

Weird. OK, it's really slow, but still less than 1 hour, and nothing we can really influence (apart from settings. I hope the compiler gets its /MP (multiprocessor build) flag?). What I don't understand is the warning C5105 about "defined" being undefined with VS2019. At least, the warning went away with VS2022. The failure is again in that unit test.

**Comment by gzotti** [comment]
Yes, but this does not make life easier :-o 
There are several warnings when compiling glues. Seems we may have to solve them!

**Comment by alex-w** [comment]
macOS:
```
FAIL!  : TestStelSphericalGeometry::testSphericalPolygon() Compared doubles are not the same (fuzzy compare)
[493](https://github.com/Stellarium/stellarium/runs/7468346906?check_suite_focus=true#step:6:494)
   Actual   (bigSquare.getUnion(holySquare)->getArea()): 0.61656327344
[494](https://github.com/Stellarium/stellarium/runs/7468346906?check_suite_focus=true#step:6:495)
   Expected (bigSquare.getArea())                      : 1.02463024879
[495](https://github.com/Stellarium/stellarium/runs/7468346906?check_suite_focus=true#step:6:496)
   Loc: [/Users/runner/work/stellarium/stellarium/src/tests/testStelSphereGeometry.cpp(368)]
```

Linux:
```
FAIL!  : TestStelSphericalGeometry::testSphericalPolygon() Compared doubles are not the same (fuzzy compare)
[500](https://github.com/Stellarium/stellarium/runs/7468346906?check_suite_focus=true#step:6:501)
   Actual   (bigSquare.getUnion(holySquare)->getArea()): 1.5
[501](https://github.com/Stellarium/stellarium/runs/7468346906?check_suite_focus=true#step:6:502)
   Expected (bigSquare.getArea())                      : 1.02463024879
[502](https://github.com/Stellarium/stellarium/runs/7468346906?check_suite_focus=true#step:6:503)
   Loc: [/home/runner/work/stellarium/stellarium/src/tests/testStelSphereGeometry.cpp(368)]
```

Windows:
```
FAIL!  : TestStelSphericalGeometry::testSphericalPolygon() Compared doubles are not the same (fuzzy compare)
[1534](https://ci.appveyor.com/project/alex-w/stellarium/builds/44250704#L1534)   Actual   (bigSquare.getUnion(holySquare)->getArea()): 1.5123151244
[1535](https://ci.appveyor.com/project/alex-w/stellarium/builds/44250704#L1535)   Expected (bigSquare.getArea())                      : 1.02463024879
[1536](https://ci.appveyor.com/project/alex-w/stellarium/builds/44250704#L1536)C:\stellarium\src\tests\testStelSphereGeometry.cpp(368) : failure location
```

WTF?!? The method `big Square.getUnion(holy Square)->getArea()` gives different results on different OS. It shouldn't be!

**Comment by gzotti** [comment]
I have seen the weirdest of effects in the last 2 days. Rsults of different lengths, different numbers etc. One time a qDebug() output "fixed" a test. Absolutely insane. 

I just started running a parallel session of debugging on one PC with Qt6.3.1 and the other Qt5.15.2. The internal structure of QVector has changed and has been combined to QList. However I still don't understand what's going on in the tessellation and SphericalRegion* code. I think the weird too large numbers reported in Qt6 are just misinterpreted addresses or random data converted to doubles. The debugger shows some internal changes and details in the QList which probably are not compatible with our existing QVector-based code. But the debugger does not show me all the fine details, and I am absolutely rookie when it comes to the debug instrumentation.

**Comment by gzotti** [comment]
Please don't care for the conflict now. I will rebase when done with my attempts.

**Comment by gzotti** [comment]
Test still fails occasionally. Must remove even more functions...

File: src/main.cpp
Comment (gzotti) [inline src/main.cpp:205]: I don't know if Mac allows this. Maybe they only have 3.3 Core profile. However we require some compatibility profile. Or Mac should really only provide a 2.1 context.

**Comment by gzotti** [comment]
The failing unit test around units and intersections of SphericalRegions is really bad, and was not able to fix the methods. However the methods are currently not used, so it need not be solved immediately. I #ifdeffed those functions away, and the program still works. Guillaume confirmed he also does not use them. 

One side-development in this branch was the program's OpenGL initialisation. ANGLE is of course no longer available. I am however unsure about the current Mac-related code. Else this can now probably be merged for first tests of 1.0beta. 

Speaking about version number,  @alex-w do you want to develop the new number scheme still in this branch or after merge?

**Review by alex-w — APPROVED** [review]
> Speaking about version number, @alex-w do you want to develop the new number scheme still in this branch or after merge?

Of course after merge

**Pull request merged**
